### PR TITLE
conda activate sets RSTUDIO_WHICH_R

### PR DESCRIPTION
**Pull request by halldc**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This PR enables RStudio to find the R executable from the active conda environment, when opening RStudio from the command-line.

### CODE REVIEW

**Comment by halldc** [comment]
@conda-forge-admin, please rerender

**Comment by halldc** [comment]
@bgruening Do you see any potential problems with this?

**Comment by bgruening** [comment]
I don't seem to be ok for me.
@halldc have you tested it on Windows?

ping @jdblischak

**Comment by halldc** [comment]
I don't have a Windows machine to test on. I've only tested on macOS.

Since this is using environment variables, perhaps it only works on Linux?

**Comment by jdblischak** [comment]
@halldc Good idea! It would be nice to make it easier for RStudio to interact with the conda-installed R. Here are my thoughts about how this will behave on the different operating systems:

**tl;dr** This PR mainly affects macOS users.

## Windows

Looking at `build.sh`, these activate and deactivate shell scripts are only executed when building for Linux and macOS. Thus they should have no effect on Windows (i.e. it will continue to use the system R when launched from the Anaconda Prompt instead of using the conda R).

And we don't want to execute them on Windows, since `export` is not defined when using the Anaconda Prompt:

```
>export RSTUDIO_WHICH_R="$CONDA_PREFIX/bin/R"
'export' is not recognized as an internal or external command,
operable program or batch file.
```

## Linux

This is already the default behavior on Linux. When I launch RStudio from the Terminal with an active conda environment, RStudio uses the conda R (verified by checking `R.home()` and `.libPaths()`).

Thus the only time these activate/deactivate scripts would have an effect on Linux is if the user had already manually set `RSTUDIO_WHICH_R`. The proposed changes in this PR would guarantee that RStudio would use the conda R (which I think is expected behavior), but the user may be surprised that `RSTUDIO_WHICH_R` is no longer set in that given Terminal after deactivating the environment.

## macOS

This PR does change the behavior on macOS. Setting `RSTUDIO_WHICH_R` to the conda R executable results in `open -a Rstudio` to open RStudio using the conda R executable instead of the system R.

**Comment by halldc** [comment]
@jdblischak Totally agree about Windows - I should've mentioned that in my reply. 

To fix the behavior you mentioned for Linux (which also affects macOS), I'll store the existing value of `RSTUDIO_WHICH_R` in a temporary environment variable so it can be restored upon deactivation.

**Comment by halldc** [comment]
@jdblischak @bgruening 
Looks like the build is failing because it's unable to connect to Github?

**Comment by jdblischak** [comment]
@halldc @bgruening I tested the latest version of the activate/deactivate scripts on both macOS and and Linux (condaforge/linux-anvil). The environment variable `RSTUDIO_WHICH_R` is properly changed when the environment is activated, and then reset to its original value when the environment is deactivated.

I support merging this PR. Good work @halldc!

**Comment by bgruening** [comment]
Thanks!

**Comment by jdblischak** [comment]
The Windows build failed due to a [permission error](https://ci.appveyor.com/project/conda-forge/r-base-feedstock/build/1.0.134#L26922):

```
PermissionError: [WinError 5] Access is denied: 'C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\tmp6quvblye\\binstar.token'
```

I assume this was sporadic, so I restarted it:

https://ci.appveyor.com/project/conda-forge/r-base-feedstock/build/1.0.135
